### PR TITLE
Refactor backing storage

### PR DESCRIPTION
**Pull request by theopolis**
This is a PoC replacement of RocksDB with SQLite as the backing store. If there are no performance penalties then removing RocksDB/Snappy from the build will:

- Decrease the binary size by approximately 2M
- Remove rocksdb / snappy / lz4 from the required link and dependency maintenance
- Enable quicker adoption into other platforms and build systems
- Decrease embedded systems requirements

### CODE REVIEW

**Comment by osqueryer** [comment]
Build origin/pr/1897/merge (Job: 2431) is good! :+1:

**Comment by theopolis** [comment]
Here are the affected benchmarks:
```
Benchmark                           Time(ns)    CPU(ns) Iterations  (RocksDB)
DATABASE_get                           25733      20116      32671  290683                                
DATABASE_store                         64053     112094       5936  125228                               
DATABASE_store_large                  137261     107360       6394  1000                               
DATABASE_store_append                 354942     277056       2536  1000                               
EVENTS_register                          599        468    1500568                                 
EVENTS_subscribe_fire                     44         64   10780343                                 
EVENTS_add_events                     531811     651253       1437  12529                               
EVENTS_retrieve_events/0/10            13265      21968      26826  277611                               
EVENTS_retrieve_events/0/50            28856      22569      28662  254196                               
EVENTS_retrieve_events/0/100           16559      22999      27946  266818                               
EVENTS_retrieve_events/0/1000          27905      21824      25879  259484    
```

For some reason the SQL implementation performs better with DATABASE_store_large, DATABASE_store_append.

File: osquery/database/benchmarks/database_benchmarks.cpp [hunk 1]
```diff
@@ -112,12 +126,14 @@ static void DATABASE_store_append(benchmark::State& state) {
...
-    setDatabaseValue(kPersistentSettings, "key" + std::to_string(k++), content);
+    setDatabaseValue(kPersistentSettings, "key" + std::to_string(k), content);
+    deleteDatabaseValue(kPersistentSettings, "key" + std::to_string(k));
+    k++;
   }
 
   // All benchmarks will share a single database handle.
   for (size_t i = 0; i < k; ++i) {
-    deleteDatabaseValue(kPersistentSettings, "key" + std::to_string(i));
+    // deleteDatabaseValue(kPersistentSettings, "key" + std::to_string(i));
```
Comment (theopolis) [inline osquery/database/benchmarks/database_benchmarks.cpp]: Get rid of this iteration.

**Comment by osqueryer** [comment]
Build origin/pr/1897/merge (Job: 2434) failed :-1:

**Comment by theopolis** [comment]
Forgot to push third-party commits ;)

**Comment by osqueryer** [comment]
Build origin/pr/1897/merge (Job: 2435) failed :-1:

**Comment by theopolis** [comment]
What does it mean!?
```
In file included from /Users/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/osx9/osquery/filesystem/darwin/plist.mm:19:
In file included from /Users/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/osx9/include/osquery/logger.h:18:
/Users/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/osx9/include/osquery/database.h:488:3:
error: 'virtual' can only appear on non-static member functions
  virtual bool check() final;
```

**Comment by osqueryer** [comment]
Build origin/pr/1897/merge (Job: 2436) is good! :+1:

**Pull request merged**